proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpMetricsUtils.java

```diff
@@ -0,0 +1,580 @@
+
+import com.wavefront.agent.handlers.ReportableEntityHandler;
+import com.wavefront.agent.preprocessor.ReportableEntityPreprocessor;
+import com.wavefront.common.MetricConstants;
+import com.wavefront.sdk.common.Pair;
+import com.wavefront.sdk.entities.histograms.HistogramGranularity;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.proto.common.v1.AnyValue;
+import io.opentelemetry.proto.common.v1.KeyValue;
+import io.opentelemetry.proto.metrics.v1.AggregationTemporality;
+import io.opentelemetry.proto.metrics.v1.ExponentialHistogram;
+import io.opentelemetry.proto.metrics.v1.ExponentialHistogramDataPoint;
+import io.opentelemetry.proto.metrics.v1.Gauge;
+import io.opentelemetry.proto.metrics.v1.Histogram;
+import io.opentelemetry.proto.metrics.v1.HistogramDataPoint;
+import io.opentelemetry.proto.metrics.v1.Metric;
+import io.opentelemetry.proto.metrics.v1.NumberDataPoint;
+import io.opentelemetry.proto.metrics.v1.ResourceMetrics;
+import io.opentelemetry.proto.metrics.v1.ScopeMetrics;
+import io.opentelemetry.proto.metrics.v1.Sum;
+import io.opentelemetry.proto.metrics.v1.Summary;
+import io.opentelemetry.proto.metrics.v1.SummaryDataPoint;
+import io.opentelemetry.proto.resource.v1.Resource;
+import wavefront.report.Annotation;
+import wavefront.report.HistogramType;
+import wavefront.report.ReportPoint;
+
+
+public class OtlpMetricsUtils {
+  public final static Logger OTLP_DATA_LOGGER = Logger.getLogger("OTLPDataLogger");
+  public static final int MILLIS_IN_MINUTE = 60 * 1000;
+  public static final int MILLIS_IN_HOUR = 60 * 60 * 1000;
+  public static final int MILLIS_IN_DAY = 24 * 60 * 60 * 1000;
+
+  public static void exportToWavefront(ExportMetricsServiceRequest request,
+                                       ReportableEntityHandler<ReportPoint, String> pointHandler,
+                                       ReportableEntityHandler<ReportPoint, String> histogramHandler,
+                                       @Nullable Supplier<ReportableEntityPreprocessor> preprocessorSupplier,
+                                       String defaultSource,
+                                       boolean includeResourceAttrsForMetrics) {
+    ReportableEntityPreprocessor preprocessor = null;
+    if (preprocessorSupplier != null) {
+      preprocessor = preprocessorSupplier.get();
+    }
+
+    for (ReportPoint point : fromOtlpRequest(request, preprocessor, defaultSource, includeResourceAttrsForMetrics)) {
+      // TODO: handle sampler
```

Comment:
Another `todo` comment, does this need to be addressed in this PR?

Reply:
Removed as there is no sampler for metrics/points.